Sroute new (#3551)
The-OpenROAD-Project/OpenROAD: src/pdn/README.md

pdngen
```

### Sroute

> The `add_sroute_connect` command is employed for connecting pins located outside of a specific power domain to the power ring, especially in cases where multiple power domains are present. During sroute, multi-cut vias will be added for new connections. The use of fixed vias from the technology file should be specified for the connection using sroute command. The use of max_rows and max_columns defines the row and column limit for the via stack. The use of hDX, hDY, vDX and vDY should be specified for finding the power ring that the pins will get connected to. 

Reviewer: ```suggestion
The `add_sroute_connect` command is employed for connecting pins located
outside of a specific power domain to the power ring, especially in cases where
multiple power domains are present. During `sroute`, multi-cut vias will be added
for new connections. The use of fixed vias from the technology file should be
specified for the connection using the `add_sroute_connect` command. The use
of max_rows and max_columns defines the row and column limit for the via stack.
The use of `hDX`, `hDY`, `vDX`, and `vDY` should be specified for finding the
power ring to which the pins will connect. 
```

Reviewer: How are "`hDX`, `hDY`, `vDX`, and `vDY`" set/given in the command?

Author: they are deleted. Thanks for noticing.

```

> "add_sroute_connect" [-net net] \
>                      [-outerNet outerNet] \
>                      -layers list_of_2_layers \
>                      -cut_pitch pitch_value \
>                      [-fixed_vias list_of_vias] \
>                      [-dont_use_vias list_of_vias] \
>                      [-max_rows rows] \
>                      [-max_columns columns] \
>                      [-stripDY stripDY] \
>                      [-metalwidths metalwidths] \
>                      [-metalspaces metalspaces] \
>                      [-ongrid ongrid_layers] \
>                      [-split_cuts split_cuts_mapping] \
>                      [-insts inst]

Reviewer: ```suggestion
add_sroute_connect
    -layers list_of_2_layers
    -cut_pitch pitch_value
    [-net net]
    [-outerNet outerNet]
    [-fixed_vias list_of_vias]
    [-dont_use_vias list_of_vias]
    [-max_rows rows]
    [-max_columns columns]
    [-stripDY stripDY]
    [-metalwidths metalwidths]
    [-metalspaces metalspaces]
    [-ongrid ongrid_layers]
    [-split_cuts split_cuts_mapping]
    [-insts inst]
```

```

#### Options

| Switch Name | Description |
| ----- | ----- |

> | `-net` | Specify the inner net where the power ring exists. |
> | `-outerNet` | Specify the outer net where instances/pins that need to get connected exist. |
> | `-layers` |  Specifying the metal layer for vertical stripe within inner power ring. |
> | `-cut_pitch` | When the two layers are parallel e.g. overlapping stdcell rails, specify the distance between via cuts |
> | `-fixed_vias` | List of fixed vias to be used to form the via stack |
> | `-dont_use_vias` | List or pattern of vias to not use to form the via stack |
> | `-max_rows` | Maximum number of rows when adding arrays of vias |
> | `-max_columns` | Maximum number of columns when adding arrays of vias |
> | `-stripDY` | Specify the width of vertical stripe inside inner power domain |
> | `-metalwidths` | Specify the width for each metal layer |
> | `-metalspaces` | Specify the spacing of each metal layer |
> | `-ongrid` | List of intermediate layers in a via stack to snap onto a routing grid |
> | `-split_cuts` | Specifies layers to use split cuts on with an associated pitch, for example `{metal3 0.380 metal5 0.500}`. |
> | `-insts` | Specifies all the instances contains the pin that need to get connected with power ring. |

Reviewer: ```suggestion
| `-net` | The inner net where the power ring exists. |
| `-outerNet` | The outer net where instances/pins that need to get connected exist. |
| `-layers` |  The metal layer for vertical stripe within inner power ring. |
| `-cut_pitch` | Distance between via cuts when the two layers are parallel, e.g., overlapping stdcell rails. |
| `-fixed_vias` | List of fixed vias to be used to form the via stack. |
| `-dont_use_vias` | List or pattern of vias to not use to form the via stack. |
| `-max_rows` | Maximum number of rows when adding arrays of vias. |
| `-max_columns` | Maximum number of columns when adding arrays of vias. |
| `-stripDY` | Width of vertical stripe inside inner power domain. |
| `-metalwidths` | Width for each metal layer. |
| `-metalspaces` | Spacing of each metal layer. |
| `-ongrid` | List of intermediate layers in a via stack to snap onto a routing grid. |
| `-split_cuts` | Layers to use split cuts on with an associated pitch, for example, `{metal3 0.380 metal5 0.500}`. |
| `-insts` | List of all the instances that contain the pin that needs to get connected with power ring. |
```
